src/test/java/eu/dissco/backend/repository/UserRepositoryIT.java

```diff
@@ -0,0 +1,86 @@
+package eu.dissco.backend.repository;
+
+import static eu.dissco.backend.TestUtils.MAPPER;
+import static eu.dissco.backend.TestUtils.USER_ID_TOKEN;
+import static eu.dissco.backend.TestUtils.givenUser;
+import static eu.dissco.backend.database.jooq.Tables.NEW_USER;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class UserRepositoryIT extends BaseRepositoryIT {
```

Comment:
Here's how I understand this: 

1) BaseRepositoryIT sets up a container to run the repository in. The BaseRepositoryIT container is based off of a library (PostgreSQLContainer.IMAGE). 
2) In this class, you can make test calls to the repository and have it behave normally, creating/updating/deleting records without impacting the database.

Reply:
Yes so the baseRepository kicks of a couple things:
- It creates a Postgres container
- It creates a datasource which can be used to interact with the container
- It checks if the database/source works with a test query
- It populates to database with the sql in the db.migrations, for this we use the Flyway library
- It create the JOOQ context object which we use in the repository

Then in this class when we start it all these things will be triggered so that when we want to test anything we will have a database with a context. We can create a repository object with an actual contact (not a mock, so it is an integration test) and start running actual methods against an actual database.

Reply:
Noted, thanks!